# PheWAS

#### 0. Preprocessing

Merge all variant calling only by the pangenome and jc by sample after rtg evaluation.
This is an example:
```
bcftools merge */fp.vcf.gz -o onlysnps_pg_allsamples.vcf.gz
bcftools merge */fn.vcf.gz -o onlysnps_jt_allsamples.vcf.gz
```
With bcftools merge we lost the information about genotype types, for recover these, we need to annotate these vcf with the multisample vcf file generated by the pangenome before do the evaluation.
```
Example command:
bcftools annotate -a tmp.2.vcf.gz -c FMT/GT easy_onlypg_allsamples.vcf.gz > easy_onlypg_allsamplesgt.vcf.gz
```
I checked if the graphs for only variants called by the pangenome and only variants called by genomic are the same. All seems okay.

AFS
![AFS](https://user-images.githubusercontent.com/52487106/220784889-a3b45a44-e25f-40cf-a39b-f80f9fe17f68.png)

Indels Length
![Lengthsmallindels](https://user-images.githubusercontent.com/52487106/220784909-d3634f3e-b8e6-4259-b022-77192efb4406.png)

AFS by positions (only variants validated by PCR)

![variants_samples](https://user-images.githubusercontent.com/52487106/219425775-b7de6905-eec4-4cc3-8ec6-06c75a6428ba.png)

![AF_bypos](https://user-images.githubusercontent.com/52487106/219425799-b3020c83-dc37-44ce-97c5-f3878e2c1894.png)

#### 1. SNPs file (only pangenome, chr12, easy region)

From the pangenome of 31 samples chr12 were called 21.596 haploid variants (only SNPs only ‘easy’ masked regions): 
- 1,373 Genotype 1
- 1,196 Genotype 0
- 9,012 Genotype missing 
12 out of 21.5k were validated in one sample (SHR) and validation rate in 60% 

#### 2. PheWAS

Working with a SNPs file with the variants called only by the pangenome. Convert flagged SNPs vcf file to chr, pos, & genotype only

```
bcftools query -f '%CHROM\t%POS\t[ %GT]\n' input.vcf.gz -o snps.easy.onlypg.nomulti.gt.txt
```

Create ".geno" file from GTonly vcf, convert to a genotype file, cut single marker of interest and run PHEWAS with subset phenotype file.

- [generategeno.R](https://github.com/Flavia95/HXB_pangenome/blob/main/scripts/generategeno.R)
- [generatepheno.R](https://github.com/Flavia95/HXB_pangenome/blob/main/scripts/generatepheno.R)

For 2 of 12 markers validated by PCR we have phenotypes association.
