# PheWAS

#### 0. Preprocessing

AFS
![AFS](https://user-images.githubusercontent.com/52487106/219425567-c0a1e862-6833-4dae-8aac-1fedb265288e.png)


Indels Length
![Lengthsmallindels](https://user-images.githubusercontent.com/52487106/219425629-4aeebafe-ae8a-43dc-97dc-9c6a5c0194f5.png)


AFS by positions (only variants validated by PCR)

![variants_samples](https://user-images.githubusercontent.com/52487106/219425775-b7de6905-eec4-4cc3-8ec6-06c75a6428ba.png)

![AF_bypos](https://user-images.githubusercontent.com/52487106/219425799-b3020c83-dc37-44ce-97c5-f3878e2c1894.png)

#### 1. SNPs file (only pangenome, easy region)


Number samples           | 31 |   
--------------| ----
number records |   21,596 | 
number multiallelic sites| 284| 

bcftools query -f '%CHROM\t%POS\t[ %GT]\n' snps_easy_onlypg_allsamples.vcf.gz -o snps_easy_onlypg_allsamples.txt

awk '{ if ($3 == "1") count++ } END { print count }' snps_easy_onlypg_allsamples.txt
1,354

awk '{ if ($3 == ".") count++ } END { print count }' snps_easy_onlypg_allsamples.txt
20,242

TO DO: 
- split multiallelics
- same samples vcf and phenotypes file


