# PheWAS

#### 0. Preprocessing

Merge all variant calling only by the pangenome by sample after rtg evaluation.
This is an example
'''
bcftools merge */fp.vcf.gz -o onlysnps_pg_allsamples.vcf.gz
bcftools merge */fn.vcf.gz -o onlysnps_jt_allsamples.vcf.gz
'''

With bcftools merge we lost the information about genotype types, for recover these, we need to annotate these vcf with the multisample vcf file generated by the pangenome before do the evaluation.
Example command:
bcftools annotate -a tmp.2.vcf.gz -c FMT/GT easy_onlypg_allsamples.vcf.gz > easy_onlypg_allsamplesgt.vcf.gz

AFS
![AFS](https://user-images.githubusercontent.com/52487106/220784889-a3b45a44-e25f-40cf-a39b-f80f9fe17f68.png)


Indels Length
![Lengthsmallindels](https://user-images.githubusercontent.com/52487106/220784909-d3634f3e-b8e6-4259-b022-77192efb4406.png)


AFS by positions (only variants validated by PCR)

![variants_samples](https://user-images.githubusercontent.com/52487106/219425775-b7de6905-eec4-4cc3-8ec6-06c75a6428ba.png)

![AF_bypos](https://user-images.githubusercontent.com/52487106/219425799-b3020c83-dc37-44ce-97c5-f3878e2c1894.png)

#### 1. SNPs file (only pangenome, easy region)


Number samples           | 31 |   
--------------| ----
number records |   21,596 | 
number multiallelic sites| 284| 

bcftools query -f '%CHROM\t%POS\t[ %GT]\n' snps_easy_onlypg_allsamples.vcf.gz -o snps_easy_onlypg_allsamples.txt

awk '{ if ($3 == "1") count++ } END { print count }' snps_easy_onlypg_allsamples.txt
1,373

awk '{ if ($3 == "0") count++ } END { print count }' snps_easy_onlypg_allsamples.txt

11,196

awk '{ if ($3 == ".") count++ } END { print count }' snps_easy_onlypg_allsamples.txt
9,012

bcftools norm -m -any snps.easy.onlypg.allsamples.gt.vcf.gz > snps.easy.onlypg.nomulti.gt.vcf.gz
this splits multiallelics.



- split multiallelics
- same samples vcf and phenotypes file


